### ISSUE TITLE
A3C example needs improvements on readability.

### ISSUE THREAD

**Issue by Phantom-Fan** [dmlc/mxnet#5195]
The current version of A3C is out-dated and have very few comments. Some of the test codes are no longer compatible with other script. Will this example be updated some day in the future? Any kind of refinement is highly appreciated. Thanks.

**Comment by loofahcus** [comment]
Maybe this pull request [add parallel actor critic algorithm #5199](https://github.com/dmlc/mxnet/pull/5199) can help.

**Comment by Phantom-Fan** [comment]
Hi, @loofahcus , thanks a lot for your sharing. It's very helpful. 

I'm trying to make it work on Vizdoom platform instead of Gym, however, the reward keeps droping batch by batch instead of increasing. I'm still trying to figure out where went wrong. 

I'm curious about how you determine the value for config.entropy_wf & vf_wf. In my experiments, I'm tuning the weights such that the out_grads are approxmately equal(abs around 1.0), is that normal?

**Comment by loofahcus** [comment]
Hi, @Phantom-Fan , glad it's useful for you. The code is just transplanted from [Minpy#parallel_actor_critic](https://github.com/dmlc/minpy/tree/master/examples/rl/parallel_actor_critic), and I think the entropy_wf & vf_wf used is suggested [here](https://github.com/muupan/async-rl/wiki), but I don't have much experience about tuning the hyper-parameters.

**Comment by Phantom-Fan** [comment]
@loofahcus Hi, I've successully adapted your code to Vizdoom platform. I think there might be some minor improvements could be done here.
[This statement](https://github.com/dmlc/mxnet/blob/master/example/reinforcement-learning/parallel_actor_critic/train.py#L83) might be too strict. The model may not be updated when t_max is set to a large value, therefore many episodes are wasted. Since you rebind the model every time when you want to update it, it's better to pass the mini-batch size to the train_step function. If so, you could update the model every time t hits t_max, making the traning process more efficient.

**Comment by loofahcus** [comment]
@Phantom-Fan I agree with you. I left it like that just for simplicity, as mentioned at [train.py#L34-L36](https://github.com/dmlc/mxnet/blob/master/example/reinforcement-learning/parallel_actor_critic/train.py#L34-L36)
